### PR TITLE
fix(SEO): improve HTML semantics for SEO MS-982

### PR DESCRIPTION
**Pull request by gajda-w**
I want to merge this change because...

# Pull Request Checklist

- [x] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx [hunk 1]
```diff
@@ -29,6 +29,8 @@ export const Navigation = ({ menu }: { menu: Maybe<Menu> }) => {
 
   return (
     <NavigationMenu
+      role="navigation"
+      aria-label="Main navigation"
```
Comment (GrzegorzDerdak) [inline apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:33]: This should also be translatable

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx [hunk 2]
```diff
@@ -29,6 +29,8 @@ export const Navigation = ({ menu }: { menu: Maybe<Menu> }) => {
 
   return (
     <NavigationMenu
+      role="navigation"
```
Comment (GrzegorzDerdak) [inline apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:32]: What tag does the NavigationMenu renders? if it's a `nav` then I'm not sure if we need the role here

File: apps/storefront/src/components/search-product-card.tsx [hunk 3]
```diff
@@ -48,36 +48,40 @@ export const SearchProductCard = ({
...
+      <Link
+        className="grid gap-2"
+        title={t(`search.go-to-product`, { name })}
+        href={paths.products.asPath({
+          slug: slug,
+        })}
+      >
+        {thumbnail ? (
+          <ProductThumbnail
+            alt={`Image of ${name}`}
```
Comment (GrzegorzDerdak) [inline apps/storefront/src/components/search-product-card.tsx:61]: I'd add also the translation for the `alt`

**Review by GrzegorzDerdak — CHANGES_REQUESTED** [review]

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx (hunk 2 above)
Comment (gajda-w) [reply apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:32]: Yes, it's nav, you are right

**Review by GrzegorzDerdak — APPROVED** [review]

**Review by soniaklimas — APPROVED** [review]
